x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java

```diff
@@ -680,4 +673,35 @@ public boolean hasReferences() {
             return refs.hasReferences();
         }
     }
+
+    public interface CreateShardContext {
+        LookupShardContext create(ShardId shardId) throws IOException;
+
+        static CreateShardContext fromSearchService(SearchService searchService) {
+            return shardId -> {
+                ShardSearchRequest shardSearchRequest = new ShardSearchRequest(shardId, 0, AliasFilter.EMPTY);
+                return LookupShardContext.fromSearchContext(
+                    searchService.createSearchContext(shardSearchRequest, SearchService.NO_TIMEOUT)
+                );
+            };
+        }
+    }
+
+    public record LookupShardContext(
+        EsPhysicalOperationProviders.ShardContext context,
+        SearchExecutionContext executionContext,
+        Releasable release
+    ) {
```

Comment:
It's not 100% clear to me that it wouldn't be better to make this an interface, but I figure I can get that in when I make the test. I just wanted to get this change in to start talking about how the test would have to change things.